src/Application/Products/Commands/UpdateProduct/UpdateProductCommandHandler.cs

```diff
@@ -11,40 +11,38 @@ namespace CleanCompanyName.DDDMicroservice.Application.Products.Commands.UpdateP
 internal class UpdateProductCommandHandler : IRequestHandler<UpdateProductCommand, ProductDto?>
 {
     private readonly IProductRepository _productRepository;
-    private readonly IDateTime _dateService;
     private readonly ILogger _logger;
     private readonly IValidator<Product> _validator;
 
     public UpdateProductCommandHandler(
         IProductRepository productRepository,
-        IDateTime dateService,
         ILogger<UpdateProductCommandHandler> logger,
         IValidator<Product> validator)
     {
         _productRepository = productRepository;
```

Comment:
I'm thinking now that maybe your domain shouldn't need to know how to get the current time.
If we want to be more specific in the domain layer we can name the field as "UtcCreatedOn" or similar, but not sure the conversion logic needs to be inside.
food for thought